### PR TITLE
Make EOS packable by Python/PyPI

### LINKED ISSUE #114: Make EOS packable by Python/PyPI
### Discussed in https://github.com/Akkudoktor-EOS/EOS/discussions/106

<div type='discussions-op-text'>

<sup>Originally posted by **b0661** October  6, 2024</sup>
I'm interested in using EOS as a library and application to be provided by [PyPI](https://pypi.org/) 

The current directory structure does not support that easily. There are some best practice directory layouts suggested in [the Python packaging tutorial](https://packaging.python.org/en/latest/tutorials/packaging-projects/). Whereas a mapping can always be configured, it is more easy and less error prone if the directory structure makes it quite clear which files will be part of a Python package.

EOS as a package name is already in use in PyPI. The package name has to be unique in [PyPI](https://pypi.org/) . Ideally the package name and the direcory name are the same.

My suggestion is to create a directory structure for the packages and move the existing files therein.

├── LICENSE
├── pyproject.toml
├── README.md
├── src/
│   └── Akkudoktor_EOS/ ------------------ the library (mostly from current modules directory)
│       ├── __init__.py
│       ├── class_akku.py
│       └── ....py
│   └── Akkudoktor_EOS_Server/ -------- the flask server
│       ├── __init__.py
│       └── flask_server.py
└── tests/
└── ...       as now, besides modules/

Maybe a short for Akkudoktor_EOS such as AEOS may be convenient.

Another benefit of packaging is that you can install the package(s) in development mode (basically a link to the source directory) and use them as an installed package (no Python PATH hassle). 

Cheers
Bobby</div>

### PR DESCRIPTION
**Pull request by b0661**
I'm interested in using EOS as a library and application to be provided by [PyPI](https://pypi.org/) 

The current directory structure does not support that easily. This pull request changes the directory layouts following best practise suggested in [the Python packaging tutorial](https://packaging.python.org/en/latest/tutorials/packaging-projects/).

As EOS as a package name is already in use in PyPI the package project name akkudoktor-eos is used. Packages shall not have a dash in Python, therefor the lengthy akkudoktoreos is used. 

The pull request moves and adapts source and test files. The directory structure is adapted to: 

├── LICENSE
├── pyproject.toml
├── README.md
├── src/
│   └── akkudoktoreos/ ------------------ the library (mostly from current modules directory)
│       ├── __init__.py
│       ├── class_akku.py
│       └── ....py
│   └── akkudoktoreosserver/ -------- the flask server
│       ├── __init__.py
│       └── flask_server.py
└── tests/
└── ...       as now, besides modules/

The pull request passed the tests and and the flask server also runs. 

Docker will not work as the Dockerfile is not adapted.

This pull request closes #114.

### Discussed in https://github.com/Akkudoktor-EOS/EOS/discussions/106

<div type='discussions-op-text'>

<sup>Originally posted by **b0661** October  6, 2024</sup>

Cheers
Bobby</div>

### CODE REVIEW

**Comment by drbacke** [comment]
We should directly create an issue due to Docker

**Review by drbacke — CHANGES_REQUESTED** [review]
Please correct the failing checks

**Comment by b0661** [comment]
> Please correct the failing checks

Corrected. Sorry did not remember to install pre-commit to my local repo.

File: pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,52 @@
...
+    "matplotlib>=3.9.2",
+    "flask>=3.0.3",
+    "scikit-learn>=1.5.2",
+    "deap>=1.4.1",
+    "joblib>=1.4.2",
+    "requests>=2.32.3",
+    "pytest>=8.3.3",
+    "pytest-cov>=5.0.0",
+    "pandas>=2.2.3",
+]
```
Comment (michaelosthege) [inline pyproject.toml:27]: Dependencies should remain in `requirements.txt`.
I am not familiar with hatchling, but when using setuptools as the build system it works like this:

https://github.com/JuBiotech/peak-performance/blob/efb527b6d811a760d55a3a6db83b0dbae13488d8/pyproject.toml#L25-L27

Comment (b0661) [reply pyproject.toml:27]: I will give setuptools a try.

Comment (b0661) [reply pyproject.toml]: Switched to setuptools.

**Comment by b0661** [comment]
> We should directly create an issue due to Docker. Also some problems to fix (@Checks).

Adapted the Dockerfile. This is untested.

@dsiebel, @michaelosthege, could you have a look on the Dockerfile. I think it has to be changed anyway as mariadb is now gone.

**Pull request merged**